src/instrument.ts

```diff
@@ -0,0 +1,249 @@
+/*!
+ * Copyright 2024 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  SEMATTRS_DB_NAME,
+  SEMATTRS_DB_STATEMENT,
+  SEMATTRS_DB_SYSTEM,
+  SEMATTRS_DB_SQL_TABLE,
+} from '@opentelemetry/semantic-conventions';
+
+import {
+  Span,
+  SpanStatusCode,
+  trace,
+  INVALID_SPAN_CONTEXT,
+  SpanAttributes,
+  TimeInput,
+  TracerProvider,
+  Link,
+  Exception,
+  SpanContext,
+  SpanStatus,
+  SpanKind,
+} from '@opentelemetry/api';
+
+const optedInPII: boolean =
+  process.env.SPANNER_ENABLE_EXTENDED_TRACING === 'true';
+
+interface SQLStatement {
+  sql: string;
+}
+
+interface observabilityOptions {
+  tracerProvider: TracerProvider;
+  enableExtendedTracing: boolean;
+}
+
+export type {observabilityOptions as ObservabilityOptions};
+
+const TRACER_NAME = 'cloud.google.com/nodejs/spanner';
+const TRACER_VERSION = '7.14.0'; // Manually hard coded, TODO: remove
+
+/**
+ * getTracer fetches the tracer from the provided tracerProvider.
+ * @param {TracerProvider} [tracerProvider] -- optional custom tracer provider
```

Comment:
nit: please remove `--` from the comments
you can check the documentation format here - https://github.com/googleapis/nodejs-spanner/blob/main/src/codec.ts#L495

Reply:
Done, thanks @alkatrivedi!